Expensify/App
# Live update pinned / unpinned chats

<If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit.>

~On hold until https://github.com/Expensify/Web-Expensify/pull/30612 is merged. This PR shouldn't break if merged before the Web-E PR, but its functionality won't work -> the `reportTogglePinned` event will never get triggered.~

No longer on hold! Web-e PR Merged

### Details
<Explanation of the change or anything fishy that is going on>

Previously, if a user toggled a pin in one device, they'd have to refresh other devices to get pin / unpin updates.

This isn't best for end users, so I added live pinning / unpinning across devices.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes https://github.com/Expensify/Expensify/issues/159214

### Tests


1. `git checkout` this & related web-e branches on your machine (or pull from master, if web-e PR was merged already)
2. Run the app on multiple local devices (ex: mobile and web)
    - Sign in to the same account on both
3. Pin a chat on one device, see the pin update on the other device

### QA Steps


1. Log in to e.cash on multiple devices (ex: desktop & web) with the same account
2. Pin a chat on one device, see the pin update live on the other device **without needing to refresh**
3. Repeat on the other device - pin or unpin, see the change update live on the original device

### Tested On


### Screenshots


#### Web


This is showing sync between web & Android:

https://user-images.githubusercontent.com/3885503/113706271-c65fb800-96de-11eb-9f12-729cf1464ba6.mov

#### Mobile Web


This is showing sync between Mobile Web (iOS - Safari) & web:

https://user-images.githubusercontent.com/3885503/113707376-1a1ed100-96e0-11eb-9ace-b4c52216ca0e.mov

#### Desktop


This is showing sync between desktop & iOS:

https://user-images.githubusercontent.com/3885503/113706840-6c132700-96df-11eb-9be4-4cd77f8c88d1.mov

#### iOS


See Desktop video ⬆️ 

#### Android


See Web video ⬆️
